ofrak_core/ofrak/resource.py

```diff
@@ -783,21 +832,25 @@ def get_most_specific_tags(self) -> Iterable[ResourceTag]:
         """
         return self._resource.get_most_specific_tags()
 
-    async def analyze_attributes(
-        self,
-        attributes_type: Type[RA],
-    ):
-        # TODO: Should we be using the version as well? The client wouldn't now the
-        #  version of the component in a client-server environment. We could do that efficiently by
-        #  adding a service method that list all available components (and their version)
-        # Check that the attributes are there to begin with
+    def _check_attributes(self, attributes_type: Type[RA]) -> Optional[RA]:
+        """
+        Try to get the current attributes.
+
+        TODO: Should we be using the version as well? The client wouldn't know the
+        version of the component in a client-server environment. We could do that efficiently by
+        adding a service method that list all available components (and their version)
```

Comment:
I don't understand this comment... What does it mean by "version"?

Reply:
It's a pretty old comment. IIRC it means that if the attributes were created by a component with a version less than the latest version, maybe we should also invalidate the attributes in that case.